### ISSUE TITLE
need limited support for servers that send null values

### ISSUE THREAD

**Issue by davepacheco** [joyent/node-fast#1]
In the 2.0.0 rewrite, I believed that it was considered illegal to send null 'data' values.  Since Node object-mode streams can't support them, I made the client detect them and treat it a protocol error.  The Moray test suite works fine with this constraint.

When I run against electric-moray, which is essentially a smart proxy for Moray, I found that electric-moray sometimes sends trailing null values in the final END message.  The reason is that it uses [this pattern](https://github.com/joyent/electric-moray/blob/93c9ab9f991ce9a6b9a6ed450146cc0d05118109/lib/server.js#L805-L812) in deciding whether to send an error:

```javascript
    var multiError = err[0] ? new verror.MultiError(err) : null;
    ...
    res.end(multiError);
```

If errors have accumulated in `err`, this sends an appropriate MultiError that wraps them, but if not, it sends a "null" data item.  It would have been better to send nothing (just call `res.send()` with no arguments) in that case, but we can't change electric-moray instances already deployed, so we need to support this at some level.

As for the scope of the problem: it looks like this only affects createBucket, updateBucket, delBucket, and sql.  Those are the only write operations that attempt to write to multiple shards, so they're the only ones with the MultiError logic.  Object-related operations only go to one shard, as do read-only bucket operations, so they don't use that code pattern.

The client implementations of createBucket, updateBucket, and delBucket don't add 'message' listeners, so they simply ignored any data received.  The client implementation of sql explicitly ignores incoming null values, in what looks like a band-aid that was added around the time electric-moray added support for the "sql" operation (and started emitting nulls).

Since the scope is so narrow, I'm leaning towards adding a per-RPC option to ignore incoming null values, and only use this option in those four RPCs.  If we find later that nulls are more commonly used, we may need to do something more invasive (like add an option to wrap all incoming values in an object so that we can properly represent null values to consumers).

**Comment by davepacheco** [comment]
This also affects the "delObject" RPC.

**Comment by davepacheco** [comment]
Fixed in v2.1.0.
